dspace-api/src/main/java/org/dspace/content/logic/LogicalStatementException.java

```diff
@@ -0,0 +1,35 @@
+/**
+ * The contents of this file are subject to the license and copyright
+ * detailed in the LICENSE and NOTICE files at the root of the source
+ * tree and available online at
+ *
+ * http://www.dspace.org/license/
+ */
+package org.dspace.content.logic;
+
+/**
+ * Exception for errors encountered while evaluating logical statements
+ * defined as spring beans.
+ *
+ * @author Kim Shepherd
+ * @version $Revision$
+ */
+public class LogicalStatementException extends Exception {
```

Comment:
hi @kshepherd, what about making this exception unchecked rather than checked? In this way the callers of the methods that throw this exception would not be forced to handle it all the time, since it might not even be clear what it is due to and how to handle it. In a more general discourse, I would prefer to use unchecked exception instead of checked because they create problems with the use of lambdas and create a coupling between the different service layers of the system. What do you think about it?

Reply:
Thanks @LucaGiamminonni , I see what you mean. I'll think about how I can refactor this, maybe best made after some of the other suggested changes

Reply:
Hi @LucaGiamminonni , sorry for the late reply to this.
i have refactored LogicalStatementException to be unchecked (by extending RuntimeException instead of Exception)
if it looks good to you, please resolve this conversation

Reply:
Hi @kshepherd , sorry for my late reply. So that's ok for me, thank you